### PR TITLE
fix unhandled transaction error

### PR DESCRIPTION
**Pull request by elpsyr**
Signed-off-by: crui <crui@relaper.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix some unhandled transaction error.
If the error caused by rollback is not captured, it will be commit successfully.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/devicetwin/dtclient/device_db.go [hunk 1]
```diff
@@ -89,58 +89,98 @@ func UpdateDeviceMulti(updates []DeviceUpdate) error {
...
+	}
 	for _, add := range adds {
 		err = SaveDevice(&add)
 
 		if err != nil {
 			klog.Errorf("save device failed: %v", err)
-			obm.Rollback()
+			errRollback := obm.Rollback()
+			if errRollback != nil {
+				klog.Errorf("transaction rollback failed: %v", err)
```
Comment (gy95) [inline edge/pkg/devicetwin/dtclient/device_db.go:104]: errRollback

File: edge/pkg/devicetwin/dtclient/device_db.go [hunk 2]
```diff
@@ -89,58 +89,98 @@ func UpdateDeviceMulti(updates []DeviceUpdate) error {
...
 		}
 	}
 
 	for _, attr := range addAttrs {
 		err = SaveDeviceAttr(&attr)
 		if err != nil {
-			obm.Rollback()
+			errRollback := obm.Rollback()
+			if errRollback != nil {
+				klog.Errorf("transaction rollback failed: %v", err)
```
Comment (gy95) [inline edge/pkg/devicetwin/dtclient/device_db.go:116]: ditto, replace err with errRollback
and the other logs need fixing

**Comment by elpsyr** [comment]
golint err with spelling mistack: deviceid to deviceID

**Review by gy95 — CHANGES_REQUESTED** [review]
seems ok

File: edge/pkg/devicetwin/dtclient/device_db.go (hunk 1 above)
Comment (elpsyr) [reply edge/pkg/devicetwin/dtclient/device_db.go:104]: oh,sorry.I will correct it right now :)

**Pull request closed without merging**